src/main/java/org/jenkinsci/plugins/workflow/cps/CpsFlowExecution.java

```diff
@@ -1951,24 +1961,35 @@ void saveExecutionIfDurable() {
      *  Key note: to avoid deadlocks we need to ensure that we don't hold a lock on this CpsFlowExecution when running saveOwner
      *   or pre-emptively lock the run before locking the execution and saving. */
     void saveOwner() {
+        /**
+         * We only want to update persistedClean if the program is over (in which case it doesn't matter), or if Jenkins
+         * is about to shut down (in which case it controls whether PERFORMANCE_OPTIMIZED Pipelines can be resumed).
+         */
+        boolean updatePersistedClean = done || Jenkins.get().isTerminating();
```

Comment:
TODO: Do we even care about changing it if the build is done? Also, I did not actually check how `Jenkins.isTerminating` works, so I should probably make sure it is true when terminators (`CpsFlowExecution.suspendAll`) are running.

Reply:
Ha, ok so `Jenkins.isTerminating` is only true _right after_ terminators run, see [Jenkins.cleanUp](https://github.com/jenkinsci/jenkins/blob/dd6c322dec260e41a0020b302378af85f79e1ee4/core/src/main/java/jenkins/model/Jenkins.java#L3417-L3419). Not a big deal, we'll just have to pass a flag to `saveOwner` when called from `suspendAll`.

This is why `FlowDurabilityTest.testDurableAgainstCleanRestartSurvivesIt` is failing.

Reply:
Changed in https://github.com/jenkinsci/workflow-cps-plugin/pull/363/commits/b360ae0cf8547bf0d56bfc36b8c39aaf10a04aa0 and https://github.com/jenkinsci/workflow-cps-plugin/pull/363/commits/eb08b5b6488961ce5c011eee774c5389a67c5d93.